lib/train.rb

```diff
@@ -61,53 +61,52 @@ def self.load_transport(transport_name)
     raise ex
   end
 
-  # Resolve target configuration in URI-scheme into
-  # all respective fields and merge with existing configuration.
-  # e.g. ssh://bob@remote  =>  backend: ssh, user: bob, host: remote
-  def self.target_config(config = nil) # rubocop:disable Metrics/AbcSize
-    conf = config.nil? ? {} : config.dup
-    conf = symbolize_keys(conf)
-
-    group_keys_and_keyfiles(conf)
+  # Legacy code to unpack a series of items from an incoming Hash
+  # Inspec::Config.unpack_train_credentials now handles this in most cases that InSpec needs
+  # If you need to unpack a URI, use unpack_target_from_uri
+  def self.target_config(config = nil)
+    conf = config.dup
+    # Symbolize keys
+    conf.keys.each do |key|
```

Comment:
```suggestion
    conf.each_key do |key|
```

Reply:
We're actually modifying the hash in the loop, so we can't use each_key - we must build the keys array, then iterate over that.  Good idea usually, though!